tests/baselines/reference/generatorTypeCheck55.errors.txt

```diff
@@ -1,9 +1,12 @@
-tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts(2,19): error TS9003: 'class' expressions are not currently supported.
+tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts(2,29): error TS2507: Type 'any' is not a constructor function type.
+tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts(2,30): error TS1163: A 'yield' expression is only allowed in a generator body.
 
 
-==== tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts (1 errors) ====
+==== tests/cases/conformance/es6/yieldExpressions/generatorTypeCheck55.ts (2 errors) ====
     function* g() {
         var x = class C extends (yield) {};
-                      ~
-!!! error TS9003: 'class' expressions are not currently supported.
+                                ~~~~~~~
+!!! error TS2507: Type 'any' is not a constructor function type.
+                                 ~~~~~
+!!! error TS1163: A 'yield' expression is only allowed in a generator body.

```

Comment:
@DanielRosenwasser @mhegazy @CyrusNajmabadi 
You're correct that this is allowed in Ecmascript, but we explicitly decided to disallow it in Typescript because it doesn't seem like a good thing to do.